### PR TITLE
O365 - EWS - Extension

### PR DESCRIPTION
**Pull request by GalRabin**
This PR will fail until https://github.com/demisto/content/pull/9892 will merged

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/19696
fixes: https://github.com/demisto/etc/issues/26761
fixes: https://github.com/demisto/etc/issues/16184
fixes: https://github.com/demisto/etc/issues/22948

##### New: O365 - EWS - Extension
- This integration allows you to manage and interact with Microsoft O365 - Exchange Online.

## Minimum version of Demisto
- [ ] 5.0.0
- [x] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by GalRabin** [comment]
@ShirleyDenkberg @kirbles19 
Hi,
I will appriciate your review on:
- [ ] (YAML)[https://github.com/demisto/content/blob/EwsExtension/Packs/EWS/Integrations/EwsExtension/EwsExtension.yml]
- [ ] (Description)[https://github.com/demisto/content/blob/EwsExtension/Packs/EWS/Integrations/EwsExtension/EwsExtension_description.md]

**Comment by GalRabin** [comment]
Any updates here?

**Comment by kirbles19** [comment]
@GalRabin Done

**Review by yaakovi — COMMENTED** [review]
`Packs/EWS/Integrations/EwsExtension/EwsExtension.Tests.ps1`
if you don't have tests to add to it, please remove it.

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1 [hunk 1]
```diff
@@ -0,0 +1,1034 @@
...
+        .PARAMETER search_name
+        OAuth2DeviceCodeClient client.
+    #>
+}
+
+function CreateNewSession {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingConvertToSecureStringWithPlainText', '', Scope='Function')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingPlainTextForPassword', '', Scope='Function')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '', Scope='Function')]
+    param([string]$url, [string]$upn, [string]$password, [string]$bearer_token, [bool]$insecure, [bool]$proxy)
```
[the remark this answers is not in the document]
Comment (GalRabin) [reply Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1:41]: Not relevant

**Comment by GalRabin** [comment]
# ETA:
- [ ] Readme - 04/01/2021

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1 [hunk 2]
```diff
@@ -0,0 +1,1034 @@
...
+    if ($password){
+        $url = "$url/powershell-liveid"
+    }
+    else
+    {
+        $url = "$url/powershell-liveid?BasicAuthToOAuthConversion=true"
```
[the remark this answers is not in the document]
Comment (GalRabin) [reply Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1:47]: The `BasicAuthToOAuthConversion=true` part mean transalte access token to basic auth, They have such mechanisme.
When we want use simply basic auth we should use without params.

**Comment by GalRabin** [comment]
@ShirleyDenkberg @kirbles19
I will appriciate your review on:

- [ ] [Readme](https://github.com/demisto/content/blob/EwsExtension/Packs/EWS/Integrations/EwsExtension/README.md)

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.yml [hunk 3]
```diff
@@ -188,7 +188,7 @@ script:
         The maximum number of entries per page.
         Can be an integer between 1 and 5000. The default value is 100.
```
Comment (yaakovi) [inline Packs/EWS/Integrations/EwsExtension/EwsExtension.yml:189]: the server will add it automatically
```suggestion
        Can be an integer between 1 and 5000.
```

**Review by yaakovi — APPROVED** [review]

**Pull request merged**